pkg/controller/kogitoinfra/kogitoinfra_resource.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kogitoinfra
+
+import (
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// InfraResource Interface to represent type of supported kogito infra resources like Infinispan, kafka & keycloak
+type InfraResource interface {
```

Comment:
You don't need to provide an interface if you can hide the functions in the packages and you will only have one implementation of it.

Reply:
`kogitoinfra_reconciler ` uses this interface reference to call reconcile method on different resources. 
https://github.com/vaibhavjainwiz/kogito-cloud-operator/blob/25ffe8183c9dfa325fa814790b8f41ab1c442834/pkg/controller/kogitoinfra/kogitoinfra_controller.go#L108

Reply:
Hi! Yes, but what I'm saying is that we do not need an interface of it's not implemented by more than once, which is the case. Plus this feature is in a package which you can hide the implementation with unexported functions.